### PR TITLE
[Merged by Bors] - feat(NormedSpace): Move toSpanNonzeroSingleton to new file and add LinearIsometryEquiv

### PR DESCRIPTION
**Pull request by mcdoll**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/NormedSpace/LinearIsometry.lean [hunk 1]
```diff
@@ -27,7 +27,8 @@ theory for `SeminormedAddCommGroup` and we specialize to `NormedAddCommGroup` wh
 
 open Function Set
 
-variable {R R₂ R₃ R₄ E E₂ E₃ E₄ F 𝓕 : Type*} [Semiring R] [Semiring R₂] [Semiring R₃] [Semiring R₄]
+variable {R R₂ R₃ R₄ R₅ E E₂ E₃ E₄ F 𝓕 : Type*}
+  [Semiring R] [Semiring R₂] [Semiring R₃] [Semiring R₄]
```
Comment (sgouezel) [inline Mathlib/Analysis/NormedSpace/LinearIsometry.lean:31]: ??

File: Mathlib/Analysis/NormedSpace/Span.lean [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+end ContinuousLinearEquiv
+
+namespace LinearIsometryEquiv
+
+variable [NormedDivisionRing 𝕜] [SeminormedAddCommGroup E] [Module 𝕜 E] [BoundedSMul 𝕜 E]
+
+/-- Given a unit element `x` of a normed space `E` over a field `𝕜`, the natural
+    linear isometry equivalence from `E` to the span of `x`.-/
+noncomputable def toSpanUnitSingleton (x : E) (hx : ‖x‖ = 1) :
+    𝕜 ≃ₗᵢ[𝕜] Submodule.span 𝕜 {x} where
```
Comment (sgouezel) [inline Mathlib/Analysis/NormedSpace/Span.lean:110]:
```suggestion
    𝕜 ≃ₗᵢ[𝕜]  𝕜 ∙ x where
```

File: Mathlib/Analysis/NormedSpace/Span.lean [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  toLinearEquiv := LinearEquiv.toSpanNonzeroSingleton 𝕜 E x (by aesop)
+  norm_map' := by
+    intro
+    rw [LinearEquiv.toSpanNonzeroSingleton_homothety, hx, one_mul]
+
+@[simp] theorem toSpanUnitSingleton_apply (x : E) (hx : ‖x‖ = 1) (r : 𝕜) :
+    toSpanUnitSingleton x hx r = (⟨r • x, by aesop⟩ : Submodule.span 𝕜 {x}) := by
```
Comment (sgouezel) [inline Mathlib/Analysis/NormedSpace/Span.lean:117]:
```suggestion
    toSpanUnitSingleton x hx r = (⟨r • x, by aesop⟩ :  𝕜 ∙ x) := by
```

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks!

File: Mathlib/Analysis/NormedSpace/LinearIsometry.lean (hunk 1 above)
Comment (mcdoll) [inline Mathlib/Analysis/NormedSpace/LinearIsometry.lean:31]:
```suggestion
variable {R R₂ R₃ R₄ E E₂ E₃ E₄ F 𝓕 : Type*} [Semiring R] [Semiring R₂] [Semiring R₃] [Semiring R₄]
```

Comment (mcdoll) [reply Mathlib/Analysis/NormedSpace/LinearIsometry.lean]: Ups, probably a rogue change from another branch. I'm sorry

**Comment by mcdoll** [comment]
bors merge

File: Mathlib/Analysis/NormedSpace/Span.lean [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    linear isometry equivalence from `E` to the span of `x`.-/
+noncomputable def toSpanUnitSingleton (x : E) (hx : ‖x‖ = 1) :
+    𝕜 ≃ₗᵢ[𝕜]  𝕜 ∙ x where
+  toLinearEquiv := LinearEquiv.toSpanNonzeroSingleton 𝕜 E x (by aesop)
+  norm_map' := by
+    intro
+    rw [LinearEquiv.toSpanNonzeroSingleton_homothety, hx, one_mul]
+
+@[simp] theorem toSpanUnitSingleton_apply (x : E) (hx : ‖x‖ = 1) (r : 𝕜) :
+    toSpanUnitSingleton x hx r = (⟨r • x, by aesop⟩ :  𝕜 ∙ x) := by
```
Comment (alreadydone) [inline Mathlib/Analysis/NormedSpace/Span.lean:117]: nit
```suggestion
    𝕜 ≃ₗᵢ[𝕜] 𝕜 ∙ x where
  toLinearEquiv := LinearEquiv.toSpanNonzeroSingleton 𝕜 E x (by aesop)
  norm_map' := by
    intro
    rw [LinearEquiv.toSpanNonzeroSingleton_homothety, hx, one_mul]

@[simp] theorem toSpanUnitSingleton_apply (x : E) (hx : ‖x‖ = 1) (r : 𝕜) :
    toSpanUnitSingleton x hx r = (⟨r • x, by aesop⟩ : 𝕜 ∙ x) := by
```

**Comment by mcdoll** [comment]
bors merge

**Pull request closed without merging**